wgengine/magicsock/magicsock.go

```diff
@@ -1523,6 +1538,14 @@ func (c *Conn) handleDiscoMessage(msg []byte, src netip.AddrPort, derpNodeSrc ke
 			c.logf("[unexpected] CallMeMaybe from peer via DERP whose netmap discokey != disco source")
 			return
 		}
+		if c.blockEndpoints {
+			c.dlogf("[v1] magicsock: disco: %v<-%v (%v, %v)  got call-me-maybe with %d endpoints, but endpoints blocked",
+				c.discoShort, epDisco.short,
+				ep.publicKey.ShortString(), derpStr(src.String()),
+				len(dm.MyNumber),
+			)
+			dm.MyNumber = []netip.AddrPort{}
```

Comment:
why are we even delivering the CallMeMaybe?  We clear out any previously learned endpoints when we set `blockEndpoints`

Reply:
This is for incoming call-me-maybe packets. If only one side has "--disable-direct-connections" then the other side will still send call-me-maybe packets populated with endpoints to the other peer, which is why they still connect directly if they are on the same LAN.

Reply:
Clearing the endpoints once won't prevent them from being readded to the peer map when the peer sends us another call-me-maybe packet.

Reply:
I realize this is for incoming call-me-maybe.  I'm asking why we strip out the endpoints and then process the _empty_ call-me-maybe, rather than just drop the call-me-maybe packets on the floor without further processing.  What benefit do we get from sending an empty call-me-maybe thru `handleCallMeMaybe()`?